HDSAviz/data_processing.py

```diff
@@ -0,0 +1,197 @@
+""" Tools for reading and processing the sensitivity analysis data files."""
+import os
+
+import pandas as pd
+
+
+def read_file(filename, numrows=None, drop=False, sep=','):
+    """
+    Function reads a file of input parameters or model results
+    and returns a pandas dataframe with its contents.
+    The first line of the input should contain headers
+    corresponding to the column names.
+
+    Parameters:
+    ----------
+    filename  : String with the complete filename, including
+                absolute or relative path.
+    numrows   : Integer number of rows of the file to read.
+                If you don't specify this parameter all rows
+                will be read.
+    drop      : A list of strings indicating which (if any)
+                of the named columns you do not want to include
+                in the resulting dataframe. (ex. ['cats', 'dogs'],
+                default is not to drop any rows)
+    sep       : String indicating the column separator in the
+                file (optional, default = ',').
+
+    Returns:
+    --------
+    df : A pandas dataframe with the contents of the file,
+         limited to the number of rows specified and without the
+         columns named in "drop".
+    """
+    df = pd.read_csv(filename, sep=sep, nrows=numrows)
+    if not drop:
+        df.drop(drop, axis=1, inplace=True)
+
+    return df
+
+
+def get_all_params(numrows=None):
+    """
+    Returns a pandas dataframe with all of the input parameters
+    set in running the model for sensitivity analysis, including
+    the parameters that are not analyzed for sensitivity (end time
+    and oxygen content).
+
+    numrows = the number of rows of the input_parameters file to read
+              (default is to read all rows).
+    """
+    return read_file('../../HDSAviz_data/input_parameters.csv',
+                     numrows=numrows)
+
+
+def get_params(numrows=None):
+    """
+    Returns a pandas dataframe with all the parameters analyzed in
+    the sensitivity analysis, but not additional parameters.
+
+    numrows = the number of rows of the input_parameters file to read
+              (default is to read all rows).
+    """
+    return read_file('../../HDSAviz_data/input_parameters.csv',
+                     numrows=numrows,
+                     drop=['End_time', 'Oxygen'])
+
+
+def get_results(numrows=None):
+    """
+    Returns a pandas dataframe with the results of running all of the
+    simulations for the parameters sets in `input_parameters.csv`. This
+    function drops two unused functional groups from the results file.
+
+    numrows = the number of rows of the input_parameters file to read
+          (default is to read all rows).
+    """
+    return read_file('../../HDSAviz_data/results.csv',
```

Comment:
Just make a note that this function and a few others are specific to the data set we are operating on and not universal (like the parameter titles to drop) that way it is clear what to change if we were to apply this to a different, unrelated data set.